framework/src/modules/interoperability/mainchain/commands/message_recovery.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { codec } from '@liskhq/lisk-codec';
+import { regularMerkleTree } from '@liskhq/lisk-tree';
+import { hash } from '@liskhq/lisk-cryptography';
+import {
+	CCM_STATUS_RECOVERED,
+	CHAIN_ACTIVE,
+	COMMAND_ID_MESSAGE_RECOVERY,
+	EMPTY_FEE_ADDRESS,
+} from '../../constants';
+import { ccmSchema, messageRecoveryParams } from '../../schema';
+import { CommandExecuteContext, VerificationResult } from '../../../../node/state_machine';
+import { MessageRecoveryParams } from '../types';
+import { CCMsg, StoreCallback } from '../../types';
+import { BaseInteroperabilityCommand } from '../../base_interoperability_command';
+import { MainchainInteroperabilityStore } from '../store';
+import { getIDAsKeyForStore } from '../../utils';
+
+export class MessageRecoveryCommand extends BaseInteroperabilityCommand {
+	public id = COMMAND_ID_MESSAGE_RECOVERY;
+	public name = 'messageRecovery';
+	public schema = messageRecoveryParams;
+
+	// TODO
+	// eslint-disable-next-line @typescript-eslint/require-await
+	public async verify(): Promise<VerificationResult> {
+		throw new Error('Method not implemented.');
+	}
+
+	public async execute(context: CommandExecuteContext<MessageRecoveryParams>): Promise<void> {
+		const { transaction, params, getAPIContext, logger, networkIdentifier, getStore } = context;
+		const apiContext = getAPIContext();
+
+		if (params.crossChainMessages.length === 0) {
```

Comment:
This should be validated in the verify hook with schema and use `minItems` to ensure there are crossChainMessages, instead of checking it explicitly here